web/docs/auth/overview.md

```diff
@@ -2,26 +2,631 @@
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```js title="src/server/actions.js"
+import HttpError from '@wasp/core/HttpError.js'
+
+export const createTask = async (task, context) => {
+  if (!context.user) {
+    throw new HttpError(403)
+  }
+
+  const Task = context.entities.Task
+  return Task.create({
+    data: {
+      description: task.description,
+      user: {
+        connect: { id: context.user.id }
+      }
+    }
+  })
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title="src/server/actions.ts"
+import type { Task } from "@wasp/entities"
+import type { CreateTask } from "@wasp/actions/types"
+import HttpError from '@wasp/core/HttpError.js'
+
+type CreateTaskPayload = Pick<Task, "description">
+
+export const createTask: CreateTask<CreateTaskPayload, Task> = async (args, context) => {
+  if (!context.user) {
+    throw new HttpError(403)
+  }
+
+  const Task = context.entities.Task
+  return Task.create({
+    data: {
+      description: args.description,
+      user: {
+        connect: { id: context.user.id }
+      }
+    }
+  })
+}
+```
+</TabItem>
+</Tabs>
+
+To implement access control, each operation is responsible for checking `context.user` and acting accordingly - e.g. if `context.user` is `undefined` and the operation is private then user should be denied access to it.
+
+When using WebSockets, the `user` object is also available on the `socket.data`. Read more in the [WebSockets](/docs/advanced/web-sockets#websocketfn-function) section.
+
+## User entity
+
+### Password hashing
+
+You don't need to worry about hashing the password yourself. Even when you are using Prisma's client directly and calling `create()` with a plain-text password, Wasp's middleware takes care of hashing it before storing it in the database.
+
+For example, if you need to update a user's password, you can do it safely via Prisma client, e.g. within an action:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```js title="src/server/actions.js"
+export const updatePassword = async (args, context) => {
+  return context.entities.User.update({
+    where: { id: args.userId },
+    data: {
+      password: 'New pwd which will be hashed automatically!'
+    }
+  })
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title="src/server/actions.ts"
+import type { UpdatePassword } from "@wasp/actions/types"
+import type { User } from "@wasp/entities"
+
+type UpdatePasswordPayload = {
+  userId: User["id"]
+}
+
+export const updatePassword: UpdatePassword<UpdatePasswordPayload, User> = async (args, context) => {
+  return context.entities.User.update({
+    where: { id: args.userId },
+    data: {
+      password: 'New pwd which will be hashed automatically!'
+    }
+  })
+}
+```
+</TabItem>
+</Tabs>
+
+### Default validations
+
+We provide basic validations out of the box, which you can customize as shown below. 
```

Comment:
Since we're using "we" as an entity going through the guide (i.e., us + the reader), we should refrain from using it when talking about Wasp.

```suggestion
Wasp includes several basic validation mechanisms. If you need something extra, the [next section](#customizing-validations) shows how to customize them.
```